HiP-DataStore/Core/HistoryUtil.cs

```diff
@@ -38,7 +39,8 @@ public static class HistoryUtil
                     baseEvent.GetEntityType() == entityId.Type && baseEvent.Id == entityId.Id)
                 {
                     var timestamp = baseEvent.Timestamp;
-                    var user = baseEvent.UserId;
+                    var userDetails = await userStoreService.Users.GetByIdAsync(baseEvent.UserId);             //retrieve the user details (according to his id) from UserResult by querying UserStore
```

Comment:
We should not query the UserStore for every event. It would be more efficient to query the data of all users once before and then filter locally.